fineract-provider/src/main/java/org/apache/fineract/useradministration/service/UserDataValidator.java

```diff
@@ -173,6 +185,11 @@ public void validateForUpdate(final String json) {
         if (this.fromApiJsonHelper.parameterExists("username", element)) {
             final String username = this.fromApiJsonHelper.extractStringNamed("username", element);
             baseDataValidator.reset().parameter("username").value(username).notBlank().notExceedingLengthOf(100);
+            // this will be cumbersome if the username remains the same. Since the parameter seems to be optional, the client can work around it...[I hope]
+            AppUser exists = appUserRepository.findAppUserByName(username);
+            if (exists != null) {
```

Comment:
This assumes the username will always be updated and will be different from the current one. In real life, the username usually stays the same when updating a user. Do the validation just for creation first and lets see if this resolves: https://issues.apache.org/jira/browse/FINERACT-803 as the issue occurs during creation.

Reply:
@awasum I'd changed the approach so we ignore the repeated username if it is the same as the current one but still flag the validation failure if the request would genuinely violate the unique constraint.

Let me just drop the validation in the update function as you've suggested.